src/data-model-providers/codegen/CodegenDataModelProvider_Read.cpp

```diff
@@ -116,27 +114,13 @@ DataModel::ActionReturnStatus CodegenDataModelProvider::ReadAttribute(const Data
     }
 
     // Read via AAI
-    std::optional<CHIP_ERROR> aai_result;
-    if (const EmberAfCluster ** cluster = std::get_if<const EmberAfCluster *>(&metadata))
```

Comment:
Is the "what is the right error if we have no metadata" thing FindAttributeMetadata does still relevant here?  Or is that all handled by callers now?

Also, the comments above this method no longer match the actual method behavior.  Those should have been adjusted in previous PRs, but we should at least do it now.

Reply:
> Is this what you are suggesting to save flash?

Yes.

Reply:
We call ReadAttribute directly in Engine.cpp and we bypass "not found":

https://github.com/project-chip/connectedhomeip/blob/master/src/app/reporting/Engine.cpp#L161

We could do this optimization and check at the IM level, however at this time just removing the check is not possible. I will remove it from write in #37322, however for read I would rather do a followup: this PR is already large and has several changes.

Reply:
Yep, if callers don't handle it, then we still need it.  I was just checking whether it's needed.